R/MsBackendTimsTof-functions.R

```diff
@@ -12,20 +12,25 @@
 #'
 #' @noRd
 .initialize <- function(x, file = character(), BPPARAM = bpparam()) {
-  L <- bplapply(seq_len(length(file)), function(fl_idx) {
-    tms <- OpenTIMS(file[fl_idx])
-    frames <- cbind(tms@frames, file = fl_idx)
-    indices <- cbind(
-      frame = rep(tms@frames$Id, tms@frames$NumScans),
-      scan = unlist(lapply(tms@frames$NumScans, seq_len)),
-      file = fl_idx)
-    #CloseTIMS(tms)
-    list(frames, indices)
-  }, BPPARAM = BPPARAM)
-  x@frames <- do.call(rbindFill, lapply(L, "[[", 1))
-  x@indices <- do.call(rbind, lapply(L, "[[", 2))
-  x@fileNames <- file
-  x
+    L <- bplapply(seq_len(length(file)), function(fl_idx) {
+        tms <- opentimsr::OpenTIMS(file[fl_idx])
+        on.exit(opentimsr::CloseTIMS(tms))
+        frames <- cbind(tms@frames, file = fl_idx)
+        indices <- cbind(
+            frame = rep(tms@frames$Id, tms@frames$NumScans),
+            scan = sequence(tms@frames$NumScans),
+            file = fl_idx)
+        list(frames, indices)
+    }, BPPARAM = BPPARAM)
+    x@frames <- do.call(rbindFill, lapply(L, "[[", 1))
+    idx <- match(colnames(x@frames, .SPECTRA_VARIABLE_MAPPINGS))
```

Comment:
the idea here is to rename some columns in `@frames` that it matches the corresponding spectra variables